script/utils/upgrades/StorageLayoutCheck.s.sol

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import { Script } from "forge-std/Script.sol";
+import { Vm } from "forge-std/Vm.sol";
+import { strings } from "solidity-stringutils/src/strings.sol";
+
+import { Utils } from "@openzeppelin-foundry-upgrades/src/internal/Utils.sol";
+import { StringUtil } from "../StringUtil.sol";
+
+/// @title StorageLayoutChecker
+/// @notice Helper contract to check if the storage layout of the contracts is compatible with upgrades
+/// @dev Picked relevant functionality from OpenZeppelin's `@openzeppelin/upgrades-core` package.
+/// NOTE: Replace this contract for their library, if these issues are resolved
+
+/// MUST be called in scripts that deploy or upgrade contracts
+/// MUST be called with `--ffi` flag
+contract StorageLayoutChecker is Script {
```

Comment:
It seems same with contract from core protocol, if so it might be better import the contract from the core protocol, so that avoid duplicate.

Reply:
Resolved in [c295705](https://github.com/storyprotocol/protocol-periphery-v1/pull/52/commits/c29570552986c317062781f72f8f435b9a6c43ec) :)